### PR TITLE
Display pictures

### PR DESCRIPTION
**Pull request by rupertlion**
## PT Link

https://www.pivotaltracker.com/story/show/160896202

## Included in PR

* Feature test for presentation of 2 images, including basic and assertion steps
* Created landing page with 2 images
* Fetching images from thecatapi.com via service in rails
* Added rest-client gem for API calls
* Added HAML-rails gem (note this is not the relevant PR - should have been separate chore - but has been included here for ease)
* Added master key and stored API key in encrypted credentials
* Moved Categories index to partial as Cats index is now the root_path
* Connecting categories and cats through before_action in cats controller
* Added FactoryBot syntax helper
* Refactored feature steps

## Screenshot

image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,4 +16,6 @@
 
 /config/master.key
 
-coverage/
\ No newline at end of file
+coverage/
+# Ignore master key for decrypting credentials and more.
+/config/master.key
```
Comment (tochman) [inline .gitignore]: See line 17

File: app/views/cats/index.html.haml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+%h1 Welcome to Cattastic!
+- @cats.each do |cat|
+    .cat
+    = image_tag cat['url']
```
Comment (tochman) [inline app/views/cats/index.html.haml]: Mind the indentation of this line if you want the img tag to be wrapped in `.cat` div. Check the rendered source in your browser.

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  root controller: :categories, action: :index
+  root controller: :cats, action: :index
+  resources :categories
```
Comment (tochman) [inline config/routes.rb]: Please scope the routes to be created with `only:`

File: .gitignore (hunk 1 above)
Comment (rupertlion) [reply .gitignore]: ah i see it - better than not having at all I guess!

File: app/views/cats/index.html.haml (hunk 2 above)
Comment (rupertlion) [reply app/views/cats/index.html.haml]: makes sense - have changed

File: config/routes.rb (hunk 3 above)
Comment (rupertlion) [reply config/routes.rb]: done

**Review by tochman — APPROVED** [review]
Looks good

**Pull request merged**